src/cartesian/Line.tsx

```diff
@@ -211,12 +211,18 @@ export class Line extends PureComponent<Props, State> {
   getStrokeDasharray = (length: number, totalLength: number, lines: number[]) => {
     const lineLength = lines.reduce((pre, next) => pre + next);
 
+    // if lineLength is 0 return the default when no strokeDasharray is provided
+    if (!lineLength) {
+      return `${length}px ${totalLength - length}px`;
+    }
+
     const count = Math.floor(length / lineLength);
     const remainLength = length % lineLength;
     const restLength = totalLength - length;
 
-    let remainLines = [];
-    for (let i = 0, sum = 0; ; sum += lines[i], ++i) {
+    let remainLines: number[] = [];
+    // prevent infinite loop by checking length https://github.com/recharts/recharts/issues/3899
```

Comment:
eh yeah. sometimes we link issues where things have slipped by in the past and adding a comment adds some awareness. This one should be pretty obvious by looking at it since it was a for-loop without a break condition 😓